src/trading/getEthFlowTransaction.ts

```diff
@@ -18,10 +18,10 @@ export async function getEthFlowTransaction(
   signer: Signer,
   appDataKeccak256: string,
   _params: LimitTradeParametersFromQuote,
+  chainId: SupportedChainId,
   networkCostsAmount = '0',
   checkEthFlowOrderExists?: EthFlowOrderExistsCallback
 ): Promise<{ orderId: string; transaction: TransactionParams }> {
-  const chainId = (await signer.getChainId()) as SupportedChainId
```

Comment:
we should not cast with `as`, we can always assert is within our list of supported chains, and then we can assume the chainId is a supporedChainId, otherwise this is not safe

I guess we can use  something like `isSupportedChainId(chainId: number): chainId is SupportedChainId`